commands/firetrade.js

```diff
@@ -1,109 +1,162 @@
+          Usage: \`!tr <Item Name>\`\n
+          Examples:
+          \`!tr Xanax\`
+          \`!tr Big Box of Chocolate Bars\`
+        `
+      }
+    });
+  }
 
-    if (itemKeys.length > 9) {
-      return message.reply('More than 9 items found. Be more precise, please!');
+  // !tr <Item Name>
+  let found_match = false;
+  let partial_matches = [];
+  let item_name = args.join(' ');
+  Object.keys(item_list_cache).some(function(item_id, index) {
+    let normalized_item_name = item_list_cache[item_id]['name']
+
+    // if an exact match is found
+    // e.g. `!tr xanex`
+    if(item_name.toUpperCase() == normalized_item_name.toUpperCase()) {
+      found_match = true
+      process(normalized_item_name)
+      return true
     }
 
-    if (itemKeys.length > 1) {
-      const content = itemKeys.length + ' items found. Choose by answering with the displayed number!';
-      let choices = '';
-      itemKeys.forEach(id => choices += '[' + id + '] ' + itemList[id].name + '\n');
-
-      return message.reply(content + '```' + choices + '```')
-        .then(() => {
-          message.channel
-            .awaitMessages(response =>
-              (response.author.username === message.author.username && itemKeys.includes(response.content)),
-              {
-                max: 1,
-                time: 10000, // 10 seconds
-                errors: ['time'],
-              }
-            )
-            .then((collected) => {
-              const selectedId = collected.first().content;
-              message.channel.send(embedFireTrading(itemList[selectedId], selectedId));
-            })
-            .catch(() => {
-              message.reply('Done waiting. You can ask me again if you still need the info.');
-            });
-        });
+    // keep track of partial matches
+    if(normalized_item_name.toUpperCase().startsWith(item_name.toUpperCase())) {
+      partial_matches.push(normalized_item_name);
     }
+  });
 
-    // Exactly one match found. Show it.
-    message.channel.send(embedFireTrading(itemList[itemKeys[0]], itemKeys[0]));
+  // if a single partial match is found
+  // e.g. `!tr xan`
+  if(!found_match && partial_matches.length == 1) {
+    found_match = true;
+    message.channel.send({
+      embed: {
+        description: `
+          Partial matches found for "${item_name}, assuming the following:
+          \`!tr ${partial_matches[0]}\`
+          `
+      }
+    })
+    process(partial_matches[0])
+  }
 
-  } catch (e) {
-    client.logger.error(`Error executing 'firetrading' command: ${e}`);
+  // if one or more partial matches are found
+  // e.g. `!tr ford`
+  if(!found_match && partial_matches.length > 1) {
+    found_match = true;
+    let matches_string = '';
+    partial_matches.forEach(match => {
+      matches_string += `\`!tr ${match}\`\n`
+    });
+
+    return message.channel.send({
+      embed: {
+        description: `
+          Partial matches found for "${item_name}, use one of the below commands:
```

Comment:
fully enclose the var with quotes. You left out the closing quote.   "${item_name}"
Same in line 118, above.